swap_meet/vendor.py

```diff
@@ -1,2 +1,87 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, id):
+        try:
+            for i in range(len(self.inventory)):
+                if self.inventory[i].id == id:
+                    return self.inventory[i]
+        except:
+            return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if their_item not in other_vendor.inventory or my_item not in self.inventory:
+            return False
+        self.inventory.append(their_item)
+        other_vendor.inventory.append(my_item)
+        other_vendor.inventory.remove(their_item)
+        self.inventory.remove(my_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False   
+
+        self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+
+        return True
+    
+    def get_by_category(self, category):
+        category_list = []
+        for element in self.inventory:
+            if element.get_category() == category:
+                category_list.append(element)
+        
+        return category_list
+
+    def get_best_by_category(self, category):
+        best_condition = 0
+        best_item = None
+        for element in self.inventory:
+            if element.get_category() == category and element.condition > best_condition:
+                best_item = element
+                best_condition = element.condition
+        return best_item
+        
+    def swap_best_by_category(self, other_vendor, my_priority, their_priority):
+        if other_vendor.get_best_by_category(my_priority) == None or self.get_best_by_category(their_priority) == None:
```

Comment:
[nit] I recommend moving this after you get `i_want` and `they_want`, so that you can use those values in the predicate and not have to call `get_best_by_category` twice.